marmoset/dhcp/dhcp_config.py

```diff
@@ -0,0 +1,68 @@
+from .isc_dhcp_ldap_config import ISCDhcpLdapConfig
+from marmoset import validation
+import socket
+
+
+class DhcpConfig:
+    def __init__(self, mac, ip_address, gateway=None, networkmask=None):
+        self.additional_statements = {}
+
+        self.mac = None
+        self.ip_address = None
+        self.gateway = None
+        self.networkmask = None
+        self.dhcp_hostname = None
+
+        self.set_settings(True, mac, ip_address, gateway, networkmask)
+
+    def set_settings(self, set_not_required_if_none=False, mac=None, ip_address=None, gateway=None, networkmask=None):
+        self.mac = mac
+
+        if gateway is not None or set_not_required_if_none:
+            self.gateway = gateway
+
+        if networkmask is not None or set_not_required_if_none:
+            self.networkmask = networkmask
+
+        if validation.is_cidr(ip_address):
+            self.ip_address = validation.get_ip_from_cidr(ip_address)
+
+            if self.networkmask is None:
+                self.networkmask = validation.get_nm_from_cidr(ip_address)
+
+            if self.gateway is None:
+                self.gateway = validation.get_gw_from_cidr(ip_address)
+        else:
+            self.ip_address = ip_address
+
+        self.dhcp_hostname = socket.getfqdn()
```

Comment:
it uses now a config value, implemented in  b3c514a20325fd971b168c57d58e4f2edebf0086  and  8cd88d37b8ac621b709c85b5b3916f3678731991